workshop/content/basics/templates/pseudo-parameters/_index.md

```diff
@@ -0,0 +1,238 @@
+
+![role-png](pseudo-parameters/role.png)
+
+You should see the IAM policy you described with your CloudFormation template. Verify that the ARN for your parameter is constructed as you expected.
+
+![policy-png](pseudo-parameters/policy.png)
+
+
+In the _Resources_ tab, you should also find the Lambda function you described with your template.
+
+To verify the Lambda function has permissions to access your SSM parameter you defined in the template, you choose to manually [invoke](https://docs.aws.amazon.com/lambda/latest/dg/getting-started-create-function.html#get-started-invoke-manually) your Lambda function. Choose to follow the link to the _Physical ID_ of your Lambda function under the _Resources_ tab of CloudFormation stack. To invoke your Lambda function for testing, you'll first need to create a Test event. On the `Test` tab, give a name to a new Test event in the new event template provided by default. Choose **Save changes** to save your Test event, and then choose **Test** to invoke your Lambda function.
+
+![lambda-test](pseudo-parameters/lambda-test.png)
+After you invoke the Lambda function, under the **Function Logs** section you should see output similar to the example shown next:
+
+![lambda-png](pseudo-parameters/lambda.png)
+
+You should see a line, similar to the one above, showing `alice` as the value for your `dbUsername` parameter: with this, you have verified that the logic you added for your Lambda function can access your parameter, and retrieve and print its value as you expected.
+
+
+### Challenge
+You’ve learned how to use pseudo parameters in your CloudFormation templates. Now let’s say you want to add an [Amazon S3 bucket](https://aws.amazon.com/s3/) to your CloudFormation template: for example, you choose the name of the S3 bucket to be as in the following format: `YOUR_BUCKET_NAME_PREFIX-AWS_REGION-YOUR_ACCOUNT_ID`, such as: `my-demo-bucket-us-east-1-111122223333`.
+
+**Task:** Describe an S3 bucket resource in your template. Add a prefix for your bucket name, and pass this prefix in as a CloudFormation template parameter. Use your template parameter and pseudo parameters to compose the bucket name as in the format mentioned earlier.
+
+{{%expand "Need a hint?" %}}
+
+- See the documentation for [CloudFormation parameters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html), and define a parameter for your S3 bucket name prefix in your template.
+- When composing the bucket name, reference your CloudFormation parameter like you referenced pseudo parameters with the `!Sub` intrinsic function earlier. For example, if your template parameter is `S3BucketNamePrefix`, choose to reference it with the `!Sub` intrinsic function as follows: `!Sub '${S3BucketNamePrefix}'`
+
+{{% /expand %}}
+
+
+
+{{%expand "Want to see the solution?" %}}
+```YAML
+# Add a template parameter that will let users input a bucket name prefix
```

Comment:
Consider breaking out this into 2 code blocks, so that the user can copy and paste from/to `Parameters` and `Resources` sections in 2 steps, and paste each section in the existing example content more easily.